utilities/tools/Set-ModuleReadMe.ps1

```diff
@@ -415,29 +423,86 @@ function Set-DeploymentExamplesSection {
+
+            # Format params with indent
+            $bicepExample = $bicepParamsArray | ForEach-Object { "  $_" }
+
+            # Optional: Add comment where required & optional parameters start
+            # ----------------------------------------------------------------
+            if ($requiredParameterNames -is [string]) {
+                $requiredParameterNames = @($requiredParameterNames)
+            }
+
+            # If we have at least one required and one other parameter we want to add a comment
+            if ($requiredParameterNames.Count -ge 1 -and $orderedJSONParameters.Keys.Count -ge 2) {
+
+                $bicepExampleArray = $bicepExample -split '\n'
+
+                # Check where the 'last' required parameter is located in the example (and what its indent is)
+                $parameterToSplitAt = $requiredParameterNames[-1]
+                $requiredParameterIndent = ([regex]::Match($bicepExampleArray[0], '^(\s+).*')).Captures.Groups[1].Value.Length
+
+                # Add a comment where the required parameters start
+                $bicepExampleArray = @('{0}// Required parameters' -f (' ' * $requiredParameterIndent)) + $bicepExampleArray[(0 .. ($bicepExampleArray.Count))]
+
+                # Find the location if the last required parameter
+                $requiredParameterStartIndex = ($bicepExampleArray | Select-String ('^[\s]{0}{1}:.+' -f "{$requiredParameterIndent}", $parameterToSplitAt) | ForEach-Object { $_.LineNumber - 1 })[0]
+
+                # If we have more than only required parameters, let's add a corresponding comment
+                if ($orderedJSONParameters.Keys.Count -gt $requiredParameterNames.Count) {
+                    $nextLineIndent = ([regex]::Match($bicepExampleArray[$requiredParameterStartIndex + 1], '^(\s+).*')).Captures.Groups[1].Value.Length
+                    if ($nextLineIndent -gt $requiredParameterIndent) {
+                        # Case Param is object/array: Search in rest of array for the next closing bracket with the same indent - and then add the search index (1) & initial index (1) count back in
+                        $requiredParameterEndIndex = ($bicepExampleArray[($requiredParameterStartIndex + 1)..($bicepExampleArray.Count)] | Select-String "^[\s]{$requiredParameterIndent}\S+" | ForEach-Object { $_.LineNumber - 1 })[0] + 1 + $requiredParameterStartIndex
+                    } else {
+                        # Case Param is single line bool/string/int: Add an index (1) for the 'required' comment
+                        $requiredParameterEndIndex = $requiredParameterStartIndex
+                    }
+
+                    # Add a comment where the non-required parameters start
+                    $bicepExampleArray = $bicepExampleArray[0..$requiredParameterEndIndex] + ('{0}// Non-required parameters' -f (' ' * $requiredParameterIndent)) + $bicepExampleArray[(($requiredParameterEndIndex + 1) .. ($bicepExampleArray.Count))]
```

Comment:
What about `additional` instead of non-required?

```suggestion
                    # Add a comment where the additional parameters start
                    $bicepExampleArray = $bicepExampleArray[0..$requiredParameterEndIndex] + ('{0}// Additional parameters' -f (' ' * $requiredParameterIndent)) + $bicepExampleArray[(($requiredParameterEndIndex + 1) .. ($bicepExampleArray.Count))]
```